src/Peer/Peer.py

```diff
@@ -157,10 +157,11 @@ def request(self, cmd, params={}, stream_to=None):
         for retry in range(1, 4):  # Retry 3 times
             try:
                 if not self.connection:
+                    # this is redundant, already established that self.connection is present
                     raise Exception("No connection found")
                 res = self.connection.request(cmd, params, stream_to)
                 if not res:
-                    raise Exception("Send error")
+                    raise Exception("Send error : result is empty")
```

Comment:
I don’t know this code well so I can only nitpick about unnecessary space before `:`.

Reply:
^_^ yeah , i shouldn't force my weird orthography into error messages